pkg/services/ngalert/schedule/schedule.go

```diff
@@ -470,15 +479,28 @@ func (sch *schedule) stopApplied(alertDefKey ngmodels.AlertRuleKey) {
 	sch.stopAppliedFunc(alertDefKey)
 }
 
-func (sch *schedule) getRuleExtraLabels(evalCtx *evaluation) map[string]string {
+func (sch *schedule) getRuleExtraLabels(rule *ngmodels.AlertRule, folderTitle string) map[string]string {
 	extraLabels := make(map[string]string, 4)
 
-	extraLabels[ngmodels.NamespaceUIDLabel] = evalCtx.rule.NamespaceUID
-	extraLabels[prometheusModel.AlertNameLabel] = evalCtx.rule.Title
-	extraLabels[ngmodels.RuleUIDLabel] = evalCtx.rule.UID
+	extraLabels[ngmodels.NamespaceUIDLabel] = rule.NamespaceUID
+	extraLabels[prometheusModel.AlertNameLabel] = rule.Title
+	extraLabels[ngmodels.RuleUIDLabel] = rule.UID
 
 	if !sch.disableGrafanaFolder {
-		extraLabels[ngmodels.FolderTitleLabel] = evalCtx.folderTitle
+		extraLabels[ngmodels.FolderTitleLabel] = folderTitle
 	}
 	return extraLabels
 }
+
+func (sch *schedule) getFolderTitle(folderTitles map[string]string, rule *ngmodels.AlertRule, missingFolders map[string][]string) string {
```

Comment:
I would recommend leaving it as it is now because it is used in only one place and it is more visible why we need missing folders map